src/Text/Markdown/SlamDown/Eval.purs

```diff
@@ -62,19 +62,19 @@ eval fs = everywhereM b i
       quoteTextBox = SD.transTextBox (runIdentity >>> SD.Literal >>> M.Just >>> Compose)
 
   f (SD.RadioButtons sel opts) = do
-    sel' <- evalExpr fs.value sel
     opts' <- evalExpr fs.list opts
-    pure $ SD.RadioButtons sel' (mergeSelection sel' opts')
+    sel' ← evalExpr (fs.value >=> \x → pure $ M.fromMaybe 0 $ L.elemIndex x (getValues opts')) sel
```

Comment:
Rather than using `fromMaybe 0`, we should try see if it is possible to restore the merging behavior here. It might be difficult though; I'll have a look at seeing if it is possible.